src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -32062,30 +32403,36 @@ static void F1<T>(Pair<T, T> p)
     }
     static void F3<T>(Pair<T, T> p) where T : class?
     {
-        var (x3, y3) = p;
-        x3.ToString(); // 3
+        var (x3, y3) = p; // 3
```

Comment:
this one seems odd-- I know CS8717 is for when `[MaybeNull]` introduces a possibly null value of an unconstrained generic type, but in this case the substituted parameters to `Deconstruct` are constrained to reference type. So the warning feels unnecessary, especially when it doesn't show up in `F2` above.

Reply:
There would be a warning for writing `default(T)` here (because `T` could be substituted with `string`), so there is a warning for producing a `[MaybeNull]T`.

---
In reply to: [324902276](https://github.com/dotnet/roslyn/pull/38610#discussion_r324902276) [](ancestors = 324902276)